### ISSUE TITLE
Configuration File v3

### ISSUE THREAD

**Issue by mageddo** [mageddo/dns-proxy-server#594]
## Summary & Motivation

After some features added, it's time to remodel and introduce a new version of the configuration file aiming a clean structure

## Goals

> What is supposed to do in short. (Optional)

Example

> A simple, static DNS server

## Non-Goals

> What people would think it is supposed to do, but it is not? (Optional)

Example
> * Not to implement a full and complex DNS server

## Description

* Depends On: #451 
* Blocks: #243 

JSON Configuration 
```
```

**Comment by defung** [comment]
I love the grouping, makes the configs more readable!

Curious, does `solver.docker.networks.preferredNetworkNames` make [my feature request](https://github.com/mageddo/dns-proxy-server/issues/595) redundant?

Also, what's `solver.docker.dpsNetwork`? Does it make DPS only resolve IPs from the `dps` network?

**Comment by mageddo** [comment]
> Curious, does solver.docker.networks.preferredNetworkNames make https://github.com/mageddo/dns-proxy-server/issues/595 redundant?

I'm considering #595 will be already implemented when this feature get done.

> Also, what's solver.docker.dpsNetwork? Does it make DPS only resolve IPs from the dps network?

[`DPS Network` and `DPS Network Auto Connect`][1] features

[1]: https://mageddo.github.io/dns-proxy-server/latest/en/3-configuration/#dps-network
